nixos/modules/services/networking/thelounge.nix

```diff
@@ -0,0 +1,77 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.thelounge;
+  dataDir = "/var/lib/thelounge";
+  configJsData = "module.exports = " + builtins.toJSON (
+    { private = cfg.private; port = cfg.port; } // cfg.extraConfig
+  );
+
+  thelounge-wrapper-script = pkgs.writeShellScriptBin "thelounge" ''
+    systemd-run --pty --property=DynamicUser=yes \
+      --property=StateDirectory=thelounge \
+      "--property=Environment=HOME=${dataDir} THELOUNGE_HOME=${dataDir}" \
+      ${pkgs.thelounge}/bin/thelounge $@
+  '';
+in {
+  options.services.thelounge = {
+    enable = mkEnableOption "The Lounge web IRC client";
+
+    private = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Make your The Lounge instance private. You will need to configure user
+        accounts by using the provided wrapper script (`thelounge`) or by adding
+        entries in <filename>${dataDir}/users</filename>. You might need to
+        restart The Lounge after making changes to the state directory.
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 9000;
+      description = "TCP port to listen on for http connections.";
+    };
+
+    extraConfig = mkOption {
+      default = {};
+      type = types.attrs;
```

Comment:
This would be a potential candidate for NixOS/rfcs#42, but that's unfortunately not yet accepted. Having the type as `types.attrs` here makes merging fail for nested values.

Reply:
I'll consider this once the RFC is accepted. I don't think this affects `thelounge` a bunch though because the surface area of configuration managed in the service definition is pretty small.